pkg/server/backuprestoreserver.go

```diff
@@ -162,6 +165,57 @@ func (b *BackupRestoreServer) runServer(ctx context.Context, restoreOpts *brtype
 	handler := b.startHTTPServer(etcdInitializer, b.config.SnapstoreConfig.Provider, b.config.EtcdConnectionConfig, nil)
 	defer handler.Stop()
 
+	//Promote self from a learner to a full member in the etcd cluster
+	for { //TODO: wise to have an infinite loop? Is there a risk of getting stuck unnecessarily here?
+		b.logger.Info("Promote loop")
+		clientFactory := etcdutil.NewFactory(brtypes.EtcdConnectionConfig{
+			Endpoints:         []string{"http://etcd-main-peer.default.svc:2380"}, //[]string{os.Getenv("ETCD_ENDPOINT")}, //TODO: Pass this via env var
+			InsecureTransport: true,
+		})
+		cli, _ := clientFactory.NewCluster()
+
+		//List all members in the etcd cluster
+		//Member URL will appear in the memberlist call response as soon as the member has been added to the cluster
+		//However, the name of the member will appear only if the member has started
+		ctx2, cancel := context.WithTimeout(context.TODO(), 5*time.Second)
+		etcdList, memListErr := cli.MemberList(ctx2)
+		cancel()
+
+		if memListErr != nil {
+			b.logger.Info("error listing members: ", memListErr)
+			continue
+		}
+
+		//TODO: Simplify logic below
+		var promoted bool
+		promoted = false
+		for _, y := range etcdList.Members {
+			b.logger.Info("CHECK: ", y.Name, " : ", os.Getenv("POD_NAME"))
+			if y.Name == os.Getenv("POD_NAME") {
+				b.logger.Info("Promoting member ", y.Name)
+				ctx2, cancel := context.WithTimeout(context.TODO(), 15*time.Second)
```

Comment:
AFAIU ctx2 will get cancel after 15sec, it means learner will only have 15sec to get in sync with leader.
Can we increase this time?